### ISSUE TITLE
TensorMemoryObj.byte_array returns dangling memoryview after allocator reuse

### ISSUE THREAD

**Issue by jduo** [LMCache/LMCache#3610]
**Label**
Please label your issue with "bug" and any other relevant labels so that it can easily be easily categorized under [LMCache Onboarding](https://github.com/LMCache/LMCache/issues/1882)

**Describe the bug**

`TensorMemoryObj.byte_array` creates a `memoryview` via `ctypes.from_address()` that has no ownership link to the underlying tensor or the MemoryObj's `ref_count`. When the allocator frees and reuses that memory region (via `ref_count_down` → `free`), the still-live memoryview becomes a dangling pointer — reading or writing through it accesses memory that now belongs to a different allocation.

This causes heap corruption when used with valkey-glide's zero-copy buffer-GET (`valkey-glide-sync >= 2.3.0`), which writes into the memoryview from a native thread without the GIL held. Manifests as segfaults, `UnicodeDecodeError` on unrelated objects, and `TypeError` on `threading.Lock` methods in vLLM EngineCore processes.

**To Reproduce**

```python
import torch
from lmcache.v1.memory_management import TensorMemoryAllocator

allocator = TensorMemoryAllocator(torch.zeros(8192, dtype=torch.uint8))

# Allocate, get byte_array, then free
obj1 = allocator.allocate(torch.Size([4096]), torch.uint8)
obj1.raw_data.fill_(0xAA)
mv1 = obj1.byte_array
assert mv1.cast("B")[0] == 0xAA

obj1.ref_count_down()  # freed — slot returned to pool

# Allocate again — gets the same slot
obj2 = allocator.allocate(torch.Size([4096]), torch.uint8)
obj2.raw_data.fill_(0xBB)

# mv1 is dangling — sees obj2's data
assert mv1.cast("B")[0] == 0xBB  # BUG: proves dangling pointer
```

**Expected behavior**

While a `byte_array` memoryview is alive, the allocator should NOT be able to recycle the underlying memory. The memoryview should keep the MemoryObj alive via `ref_count`.

**Root Cause**

In [`TensorMemoryObj.byte_array`](https://github.com/LMCache/LMCache/blob/v0.4.3/lmcache/v1/memory_management.py#L644-L662):

```python
byte_array = (ctypes.c_ubyte * num_bytes).from_address(
    ctypes.addressof(ubyte_ptr.contents)
)
return memoryview(byte_array)
```

The ctypes array created by `from_address()` does not hold a reference to `self.raw_data` (the tensor). When [`ref_count_down`](https://github.com/LMCache/LMCache/blob/v0.4.3/lmcache/v1/memory_management.py#L539-L555) triggers `allocator.free(self)`, the address range is marked free and reused immediately, but any previously-returned memoryview still points to it.

**Suggested Fix**

Bump `ref_count` in `byte_array` and register a `weakref.finalize` on the ctypes array to decrement when collected:

```python
byte_array._tensor_ref = self.raw_data
self.ref_count_up()
weakref.finalize(byte_array, self.ref_count_down)
```

PR with fix and regression test: (link to PR once created)

**Desktop (please complete the following information):**

- OS: Amazon Linux 2023 (EC2 g6.12xlarge)
- Python: 3.12
- LMCache: 0.4.3
- vLLM: 0.21.0
- valkey-glide-sync: 2.3.1

**Additional context**

- Reported originally at https://github.com/valkey-io/valkey-glide/issues/6215
- The bug is masked when using valkey-glide < 2.3.0 because the non-buffer GET path copies data with the GIL held, which serializes access to the buffer
- The corruption is sporadic (minutes to hours under load) because it requires the allocator to recycle a slot during the narrow window of a GIL-released FFI write

**Comment by jduo** [comment]
This should be labelled "bug", but I can't seem to edit labels.
